cc_plugin_ugrid/checker_10.py

```diff
@@ -120,21 +121,69 @@ def check_node_coordinates_exist(self, ds):
             return None
         except AttributeError:
             # No node_dimensions attribute... there are larger issues
+            self.node_dim = None
             return None
         except AssertionError:
             m = '"node_coordinates" member "{}" is not a variable'.format(n)
             messages.append(m)
+            self.node_dim = None
+        else:
+            # verify that the node coordinates' dimensions are equivalent
+            node_coord = n
+            try:
+                for ncoord in ncs:
+                    assert ds[ncoord].dimensions == ds[node_coord].dimensions
+            except AssertionError:
+                # names are not equivalent
+                m = 'names of node coordinates appear to be different'
```

Comment:
I feel like my response to this message would be "oh, do they now?", lets make an assertion:

node_coordinate values must match.....though I'm uncertain what we're matching.